curriculum/challenges/english/20-upcoming-python/learn-algorithm-design-by-building-the-shortest-path-algorithm/65579228c669fcbebffd01d5.md

```diff
@@ -0,0 +1,58 @@
+---
+id: 65579228c669fcbebffd01d5
+title: Step 47
+challengeType: 20
+dashedName: step-47
+---
+
+# --description--
+
+The other bug is subtle. When a shorter distance is found for a neighbor node, `paths[current]` gets assigned to the neighbor node path, `paths[node]`. This means both variables point to the same list. Since lists are mutable, when you append the neighbor node to its path, both `paths[node]` and `paths[current]` are modified because they are the same list. This results in wrong paths, although the distances are correct.
```

Comment:
```suggestion
The other bug is subtle. When a shorter distance is found for a neighbor node, `paths[current]` gets assigned to the neighbor node path, `paths[node]`. 

This means both variables point to the same list. Since lists are mutable, when you append the neighbor node to its path, both `paths[node]` and `paths[current]` are modified because they are the same list. This results in wrong paths, although the distances are correct.
```